CatAnalyzer/python/analyzers/ntuple_cff.py

```diff
@@ -0,0 +1,161 @@
+import FWCore.ParameterSet.Config as cms
+
+def ntupler_load(process, name):
+    process.load("CATTools.CatAnalyzer.analyzers.ntuple_cfi")
+    for typeName in ['int', 'ints', 'float', 'floats']:
+        for key, val in getattr(process.ntuple, typeName).parameters_().iteritems():
+            setattr(getattr(process.ntuple, typeName), key, cms.InputTag(val.value().replace('eventSel', name)))
+    for key, val in process.ntuple.cands.parameters_().iteritems():
+        getattr(process.ntuple.cands, key).src = cms.InputTag(val.src.value().replace('eventSel', name))
+    return process
+
+def ntupler_addVarsTTLL(process, name):
+    #process.load("CATTools.CatAnalyzer.ttll.ttbarDileptonKinSolutionProducer_cfi")
+    from CATTools.CatAnalyzer.ttll.ttbarDileptonKinSolutionProducer_cfi import ttbarDileptonKin
+    from CATTools.CatAnalyzer.ttll.ttbarDileptonKinSolutionAlgos_cff import ttbarDileptonKinAlgoPSetDESYSmeared as solverAlgo
+    process.ttllKin = ttbarDileptonKin.clone()
+    del ttbarDileptonKin
+    process.ttllKin.solver = solverAlgo
+
+    process.RandomNumberGeneratorService = cms.Service("RandomNumberGeneratorService",
+        ttllKin = cms.PSet(
+            initialSeed = cms.untracked.uint32(123456),
+            engineName = cms.untracked.string('TRandom3')
+        )
+    )
+
+    process.ttllLeptons = cms.EDProducer("CandViewMerger",
+        src = cms.VInputTag(cms.InputTag(name, "electrons"), cms.InputTag(name, "muons")),
+    )
+    process.ttllKin.jets = name+":jets"
+    process.ttllKin.leptons = "ttllLeptons"
+    process.ttllKin.met = name+":met"
+    process.ttllKin.metphi = name+":metphi"
+
+    #process.ntuple.floats.weights_genOthers  = cms.InputTag("flatGenWeights", "others")
+    process.ntuple.floats.kin_mLL = cms.InputTag("ttllKin", "mLL")
+    process.ntuple.floats.kin_mLB = cms.InputTag("ttllKin", "mLB")
+    #process.ntuple.floats.kin_mAddJJ = cms.InputTag("ttllKin", "mAddJJ")
+    #process.ntuple.floats.kin_dphi = cms.InputTag("ttllKin", "dphi")
+    process.ntuple.floats.kin_quality = cms.InputTag("ttllKin", "quality")
+
+    process.ntuple.cands.kin = cms.PSet(
+        src = cms.InputTag("ttllKin"),
+        exprsF = cms.untracked.PSet(
+            pt  = cms.string("pt"),
+            eta = cms.string("eta"),
+            phi = cms.string("phi"),
+            m   = cms.string("mass"),
+        ),
+    )
+    return process
+
+def ntupler_addVarsTTGen(process):
```

Comment:
This part is the beginning of working example to add partonTop and pseudoTop in the flat ntuple.